config/runtime.exs

```diff
@@ -88,27 +88,43 @@ if config_env() == :prod do
       """)
   end
 
-  # hCaptcha
-  hcaptcha_site_key = System.get_env("HCAPTCHA_SITE_KEY")
-  hcaptcha_secret_key = System.get_env("HCAPTCHA_SECRET_KEY")
-  hcaptcha_url = System.get_env("HCAPTCHA_URL")
+  # Captcha
+  captcha_site_key = System.get_env("CAPTCHA_SITE_KEY") || System.get_env("HCAPTCHA_SITE_KEY")
 
-  if hcaptcha_site_key not in [nil, ""] and hcaptcha_secret_key not in [nil, ""] do
-    config =
-      [secret_key: hcaptcha_secret_key, site_key: hcaptcha_site_key]
-      |> put_if_not_empty.(:url, hcaptcha_url)
+  captcha_secret_key =
+    System.get_env("CAPTCHA_SECRET_KEY") || System.get_env("HCAPTCHA_SECRET_KEY")
+
+  captcha_url = System.get_env("CAPTCHA_URL") || System.get_env("HCAPTCHA_URL")
+
+  if captcha_site_key not in [nil, ""] and captcha_secret_key not in [nil, ""] do
+    captcha_type = if captcha_site_key =~ "-", do: "hcaptcha", else: "friendlycaptcha"
```

Comment:
I don’t think this method is suitable for determining the captcha type. I suggest you use an additional `CAPTCHA_PROVIDER` env variable for that.

Reply:
So I think you should do something like this:

```elixir
captcha_provider =
  System.get_env("CAPTCHA_PROVIDER", "hcaptcha")
  |> String.downcase()
  |> case do
    "friendly_captcha" -> :friendly_captcha
    _other -> :hcaptcha
  end
end
```

Reply:
Good idea!